stacks-signer/src/monitoring/mod.rs

```diff
@@ -128,8 +128,9 @@ pub fn new_rpc_call_timer(_full_path: &str, _origin: &str) -> NoOpTimer {
 /// a given block. The block's timestamp is used to calculate the latency.
 ///
 /// Call this right after broadcasting a BlockResponse
-#[allow(unused_variables)]
 pub fn record_block_response_latency(block: &NakamotoBlock) {
+    #[cfg(not(feature = "monitoring_prom"))]
```

Comment:
Please don't pepper the code base with feature macros.  Instead, please consolidate feature-gated code into its own module (or method) and ship a no-op module (or method) that gets compiled when the feature is not enabled.

Reply:
Refactored in https://github.com/stacks-network/stacks-core/pull/5664/commits/4fcb4f111dc706a2c48f4eb98f1838c7924cb687, let me know what you think!